application/classes/Ushahidi/Formatter/Post/CSV.php

```diff
@@ -81,15 +83,37 @@ protected function generateCSVRecords($records)
 
 	private function getValueFromRecord($record, $keyParam){
 		$return = '';
-		$keySet = explode('.', $keyParam); //contains key + index of the key, if any
```

Comment:
ugh . this is awful. Looking to figure out something less terrible if I can.

Reply:
I think we should have a CSV format field that drives how they are shown in a csv.  Right now it's ridiculous.

Reply:
ok my comment is outdated. This is still awful BUT now I removed the extra awfulness that I introduced with the single column arrays (categories in a single column instead of many, for instance ) and made it depend on an array of pre-defined formats by key/value pair  that depend on field type. It's still far from nice, but it's a step in the right direction I believe

Reply:
removing outdated comment